core/src/main/java/com/linecorp/armeria/common/metric/DropwizardMeterRegistry.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright 2017 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package com.linecorp.armeria.common.metric;
+
+import static java.util.Comparator.comparing;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Iterator;
+
+import com.google.common.collect.Streams;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.NamingConvention;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.util.HierarchicalNameMapper;
+
+/**
+ * {@link MeterRegistry} implementation for <a href="http://metrics.dropwizard.io/">Dropwizard Metrics</a>.
+ * This implementation adds more convenient constructors and sets more sensible default {@link NamingConvention}
+ * and {@link HierarchicalNameMapper} on top of the upstream implementation.
+ */
+@SuppressWarnings("ClassNameSameAsAncestorName")
+public class DropwizardMeterRegistry extends io.micrometer.core.instrument.dropwizard.DropwizardMeterRegistry {
```

Comment:
Ah I didn't mean combining them into the same class. But instead of using inheritance, I think this could just be simple factories like this, so no need to worry about the `ClassNameSameAsAncestorName` that can be a bit confusing.

```java
public final class DropwizardMeterRegistries {
  public static DropwizardMeterRegistry default() {
    return of(DEFAULT_NAME_MAPPER);
  }

  public static DropwizardMeterRegistery of(nameMapper) {
    return of(nameMapper, Clock.System);
  }

  public static DropwizardMeterRegistry of(nameMapper, clock) {
    DropwizardMeterRegistry registry = new DropwizardMeterRegistry(nameMapper, clock);
    registry.config().namingConvention(MoreNamingConventions.dropwizard());
  }
}
```